tests/DCPS/XTypes/XTypes.cpp

```diff
@@ -0,0 +1,670 @@
+      } else if (type == "AdditionalPostfixFieldStruct") {
+        write_additional_postfix_field_struct(dw);
+      } else if (type == "MutableStruct") {
+        write_mutable_struct(dw);
+      } else if (type == "ModifiedMutableStruct") {
+        write_modified_mutable_struct(dw);
+      } else if (type == "MutableUnion") {
+        write_mutable_union(dw);
+      } else if (type == "ModifiedMutableUnion") {
+        write_modified_mutable_union(dw);
+      } else if (type == "Trim64Struct") {
+        write_trim64_struct(dw);
+      } else if (type == "AppendableStructWithDependency") {
+        write_appendable_struct_with_dependency(dw);
+      }
+    }
+
+    ACE_OS::sleep(ACE_Time_Value(3, 0));
+
+  } else if (reader) {
+    ACE_DEBUG((LM_DEBUG, "Reader starting at %T\n"));
+
+    Subscriber_var sub = dp->create_subscriber(SUBSCRIBER_QOS_DEFAULT, 0,
+                                               DEFAULT_STATUS_MASK);
+    DataReaderQos dr_qos;
+    sub->get_default_datareader_qos(dr_qos);
+    dr_qos.reliability.kind = RELIABLE_RELIABILITY_QOS;
+    dr_qos.durability.kind = TRANSIENT_LOCAL_DURABILITY_QOS;
+
+    DataReader_var dr = sub->create_datareader(topic, dr_qos, 0,
+      DEFAULT_STATUS_MASK);
+
+    ACE_OS::sleep(ACE_Time_Value(1, 0));
+
+    failed = !check_inconsistent_topic_status(topic);
+
+    if (failed) {
+      ACE_ERROR((LM_ERROR, "ERROR: Reader failed for type %s\n", type.c_str()));
+    } else if (expect_to_match) {
+      if (type == "PlainCdrStruct") {
+        failed = !(read_plain_cdr_struct(dr) == RETCODE_OK);
+      } else if (type == "AppendableStruct") {
+        failed = !(read_appendable_struct(dr) == RETCODE_OK);
+      } else if (type == "FinalStruct") {
+        failed = !(read_final_struct(dr) == RETCODE_OK);
+      } else if (type == "ModifiedFinalStruct") {
+        failed = !(read_modified_final_struct(dr) == RETCODE_OK);
+      } else if (type == "AdditionalPrefixFieldStruct") {
+        failed = !(read_additional_prefix_field_struct(dr) == RETCODE_OK);
+      } else if (type == "AdditionalPostfixFieldStruct") {
+        failed = !(read_additional_postfix_field_struct(dr) == RETCODE_OK);
+      } else if (type == "MutableStruct") {
+        failed = !(read_mutable_struct(dr) == RETCODE_OK);
+      } else if (type == "ModifiedMutableStruct") {
+        failed = !(read_modified_mutable_struct(dr) == RETCODE_OK);
+      } else if (type == "MutableUnion") {
+        failed = !(read_mutable_union(dr) == RETCODE_OK);
+      } else if (type == "ModifiedMutableUnion") {
+        failed = !(read_modified_mutable_union(dr) == RETCODE_OK);
+      } else if (type == "Trim20Struct") {
+        failed = !(read_trim20_struct(dr) == RETCODE_OK);
+      } else if (type == "AppendableStructWithDependency") {
+        failed = !(read_appendable_struct_with_dependency(dr) == RETCODE_OK);
+      }
+    }
+  } else {
+    ACE_ERROR((LM_ERROR, "ERROR: Must pass either --writer or --reader\n"));
+    failed = true;
+  }
```

Comment:
Because of the newly added check above, this won't be reached and can be deleted.